AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1326 @@
+	}
+
+	start_time_span = az_span_create((uint8_t*)incoming_since_value, incoming_since_value_len);
+
+	// Set the response payload to error if the "since" field was not sent
+	if (az_span_ptr(start_time_span) == NULL)
+	{
+		RETURN_ERR_IF_FAILED(build_error_response_payload(response, span_error_no_payload_getmaxmin, out_response));
+		return AZ_ERROR_ITEM_NOT_FOUND;
+	}
+
+	// Get the current time as a string
+	time_t rawtime;
+	struct tm* timeinfo;
+	time(&rawtime);
+	timeinfo = localtime(&rawtime);
+	size_t len = strftime(end_time_buffer, sizeof(end_time_buffer), iso_spec_time_format, timeinfo);
+	az_span end_time_span = az_span_create((uint8_t*)end_time_buffer, (int32_t)len);
+
+	RETURN_ERR_IF_FAILED(build_getMaxMinReport_response_payload(
+								response,
+								start_time_span,
+								end_time_span,
+								out_response));
+
+	return AZ_OK;
+
 }
 
-static int send_telemetry_message(void)
+/**********************************************
+*	Handle reboot command
+**********************************************/
+static az_result process_reboot(az_span payload, az_span response, az_span* out_response)
 {
-  int rc = az_iot_pnp_client_telemetry_get_publish_topic(
-              &pnp_client, AZ_SPAN_EMPTY, NULL, telemetry_topic, sizeof(telemetry_topic), NULL);
-  RETURN_ERR_WITH_MESSAGE_IF_FAILED(rc, "Could not get the telemetry topic");
+	char reboot_delay[32];
+	char buffer[32] = {0};
+	az_json_writer jw;
+	az_json_reader jr;
+
+	debug_printInfo("  MAIN: %s() : Payload %s", __func__, az_span_ptr(payload));
+
+	if (az_span_size(payload) == 0 || (az_span_size(payload) == 2 && az_span_is_content_equal(empty_payload_span, payload)))
+	{
+		RETURN_ERR_IF_FAILED(build_error_response_payload(response, span_error_no_payload_reboot, out_response));
+		return AZ_ERROR_ITEM_NOT_FOUND;
+	}
+
+	RETURN_ERR_IF_FAILED(az_json_reader_init(&jr, payload, NULL));
+
+	while (jr.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+	{
+		if (jr.token.kind == AZ_JSON_TOKEN_PROPERTY_NAME)
+		{
+			if (az_json_token_is_text_equal(&jr.token, span_command_reboot_delay))
+			{
+				RETURN_ERR_IF_FAILED(az_json_reader_next_token(&jr));
+				RETURN_ERR_IF_FAILED(az_json_token_get_string(&jr.token, reboot_delay, sizeof(reboot_delay), NULL));
+				break;
+			}
+		}
+		else if (jr.token.kind == AZ_JSON_TOKEN_STRING)
+		{
+				RETURN_ERR_IF_FAILED(az_json_token_get_string(&jr.token, reboot_delay, sizeof(reboot_delay), NULL));
+				break;
+		}
+		RETURN_ERR_IF_FAILED(az_json_reader_next_token(&jr));
+	}
+
+	if (reboot_delay[0] != 'P' || reboot_delay[1] != 'T' || reboot_delay[strlen(reboot_delay) - 1] != 'S')
+	{
+		debug_printError("  MAIN: Reboot Delay wrong format");
+		RETURN_ERR_IF_FAILED(build_error_response_payload(response, span_error_no_payload_reboot, out_response));
+		return AZ_ERROR_ARG;
+	}
+
+	memcpy(buffer, &reboot_delay[2], strlen(reboot_delay) - 3);
+	buffer[strlen(reboot_delay) - 3] = '\0';
+	rebootDelay = atoi((const char *)buffer);
```

Comment:
you can just do

```suggestion
	rebootDelay = atoi((const char *)&reboot_delay[2]);
```
atoi will stop on the first non-numeric value